lib/index.js

```diff
@@ -1042,13 +1039,45 @@ module.exports = {
             var expectedAttributeValue =
               expectedValueByAttributeName[attributeName];
             promiseByKey.attributes[attributeName] = expect.promise(function() {
-              if (expectedAttributeValue === true) {
+              if (typeof expectedAttributeValue === 'undefined') {
                 return bubbleError(function() {
-                  expect(subject.hasAttribute(attributeName), 'to be true');
+                  expect(subject.hasAttribute(attributeName), 'to be false');
                 });
-              } else if (typeof expectedAttributeValue === 'undefined') {
+              } else if (isEnumeratedAttribute(attributeName)) {
                 return bubbleError(function() {
-                  expect(subject.hasAttribute(attributeName), 'to be false');
+                  var indexOfEnumeratedAttributeValue = enumeratedAttributeValues[
+                    attributeName
+                  ].indexOf(expectedAttributeValue);
+                  if (indexOfEnumeratedAttributeValue === -1) {
+                    expect.withError(
+                      function() {
+                        expect(
+                          enumeratedAttributeValues[attributeName],
+                          'to contain',
+                          expectedAttributeValue
+                        );
+                      },
+                      function(e) {
+                        expect.fail({
+                          message: function(output) {
+                            output
+                              .text('Invalid expected value ')
+                              .appendInspected(expectedAttributeValue)
+                              .text('. Supported values include: ')
+                              .appendInspected(
+                                enumeratedAttributeValues[attributeName]
+                              );
+                          }
```

Comment:
Is it really necessary to state the supported values?

Reply:
If we are going to do this, we shoudn't use withError inside bubbleError as bubbleError is just sugar for withError. We should just set the error mode on the error directly.

Reply:
Sorry - I see this is just another sanity check. Then we should just change the bubbleError code.